src/Listener/CreditMemoGeneratedEventListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Listener;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\RefundPlugin\Entity\CreditMemoInterface;
+use Sylius\RefundPlugin\Event\CreditMemoGenerated;
+use Sylius\RefundPlugin\Exception\CreditMemoNotFound;
+use Sylius\RefundPlugin\Exception\OrderNotFound;
+use Sylius\RefundPlugin\Sender\CreditMemoEmailSenderInterface;
+
+final class CreditMemoGeneratedEventListener
+{
+    /** @var RepositoryInterface */
+    private $creditMemoRepository;
+
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var CreditMemoEmailSenderInterface */
+    private $creditMemoEmailSender;
+
+    public function __construct(
+        RepositoryInterface $creditMemoRepository,
+        OrderRepositoryInterface $orderRepository,
+        CreditMemoEmailSenderInterface $creditMemoEmailSender
+    ) {
+        $this->creditMemoRepository = $creditMemoRepository;
+        $this->orderRepository = $orderRepository;
+        $this->creditMemoEmailSender = $creditMemoEmailSender;
+    }
+
+    public function __invoke(CreditMemoGenerated $event): void
+    {
+        /** @var CreditMemoInterface|null $creditMemo */
+        $creditMemo = $this->creditMemoRepository->findOneBy(['number' => $event->number()]);
+        if ($creditMemo === null) {
+            throw CreditMemoNotFound::withNumber($event->number());
+        }
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneByNumber($event->orderNumber());
+        if ($order === null) {
+            throw OrderNotFound::withOrderNumber($event->orderNumber());
+        }
+
+        assert($order->getCustomer() !== null);
```

Comment:
Could we change that to a Webmozart/Assert call or an if? `assert` functions can be actually disabled in the runtime and PHP docs states:

> Assertions should be used as a debugging feature only. You may use them for sanity-checks that test for conditions that should always be TRUE and that indicate some programming errors if not or to check for the presence of certain features like extension functions or certain system limits and features.

http://php.net/manual/pl/function.assert.php

Reply:
Sure!